### DISCUSSION TITLE
generate ObjectId before repo.create()

### DISCUSSION THREAD

**Discussion by bebjakub** [mikro-orm/mikro-orm#4395] [Q&A] [answered]
Hello,
I need to know the document ObjectId before inserting to DB. I'm using this approach, is it correct?

``` 
const donationId = new ObjectId().toHexString();
const stripePaymentIntent = await this.stripeService.createIntent({
      amount,
      currency: 'eur',
      metadata: {
            collection: 'donation',
            id: donationId,
      },
});
const donation = this.repo.create({
      id: donationId,
      ...donationData,
      stripe: stripePaymentIntent
})

await this.em.persistAndFlush(donation).catch(error => { console.log(error) })

return donation
```

It is working, but when Iḿ using an already existing id, the document insert is skipped, but no error is thrown. so it returns an already existing donation.

Thank you!

**Comment by B4nan** [comment, marked as answer]
Yes, this is expected behavior, `em.create` will prefer returning existing entities from context if you pass in the PK. Sounds a bit weird that you would get a collision on `new ObjectId()` though. Note that you can just put that to the property initializer.

```ts
@PrimaryKey()
_id: ObjectId = new ObjectId();
```

**Reply by bebjakub** [reply]
I had a hardcoded ObjectId for testing purposes, that was the reason for the collision.
